src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -5804,6 +5828,74 @@ public enum EventManifestOptions
         AllowEventSourceOverride = 0x8,
     }
 
+#if FEATURE_EVENTSOURCE_XPLAT 
+
+    internal  class XplatEventLogger : EventListener
```

Comment:
Certainly for Unix operating systems this seems like a good plan (though the underlying mechanism will be different).  For non-Unix, we'll have to see if the same plan will work at the time.